app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+
+        Args:
+            full_url (str): The full url of GET request.
+            headers (dict): A dictionary includes bearer token.
+
+        Returns:
+            bool: True if GET response is valid.
+            dict/string/None: If GET request to fetch log files and support API
+                version list, it will return a dict. If it requests to fetch event
+                logs, it will return raw text which can be decoded by CSV DictReader
+                later. Otherwise it will return None.
+        """
+        @backoff.on_exception(backoff.expo,
+                              self.EXCEPTIONS_TO_BACKOFF,
+                              max_tries=self.BACKOFF_MAX_RETRIES)
+        def _make_get_request():
+            # To use closure here is to make backoff logic patchable and testable.
+            try:
+                # Log GET request URL for debugging purpose, especially useful when
+                # debugging query syntax
+                LOGGER.debug('URL of GET request is %s', full_url)
+                resp = requests.get(full_url, headers=headers, params=params, timeout=self._TIMEOUT)
+
+                # Return false if resp contains non-200 status code.
+                if not self._validate_status_code(resp):
+                    return False, None
+
+                # When querying list of api versions and log files, Salesforce responses
+                # json content.
+                return True, resp.json()
+            except requests.exceptions.Timeout:
+                LOGGER.exception('Request timed out for when sending get request to %s', full_url)
+                return False, None
+            except ValueError:
+                # When fetch log events, Salesforce returns raw data in csv format, not json
+                return True, resp.text.encode('utf-8')
+
+        return _make_get_request()
+
+    def _get_latest_api_version(self):
+        """GET request to fetch supported API versions and find the latest API version
+
+        The example of response json body:
+        [
+            {
+                "version": "20.0",
+                "label": "Winter '11",
+                "url": "/services/data/v20.0"
+            },
+            {
+                "version": "21.0",
+                "label": "Spring '11",
+                "url": "/services/data/v21.0"
+            },
+            {
+                "version": "26.0",
+                "label": "Winter '13",
+                "url": "/services/data/v26.0"
+            }
+        ]
+        """
+        # instance_url identifies the Salesforce instance to which API calls should be sent.
+        if not self._instance_url:
+            LOGGER.error('No instance url found for %s %s', self.service, self._type)
+            return
+
+        url = '{}/services/data/'.format(self._instance_url)
+        success, response = self._make_get_request(url, self._auth_headers)
+
+        if not success:
```

Comment:
make this `if not (success and response):...` and then you can remove the `if response` check below